File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        Map<Integer , Boolean> mp = new HashMap<>();
        
        for(int num : nums) {
            if(num > 0 && num <= nums.length) {
                mp.put(num , true);
            }
        }
        
        int i = 1;
        while(true) {
            if(mp.containsKey(i))
                i++;
            else return i;
        }
    }
}

File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        map<int,bool> mp;
        
        for(int num : nums) {
            if(num > 0 && num <= nums.size()) {
                mp[num] = 1;
            }
        }
        
        int i = 1;
        while(1) {
            if(mp[i])
                i++;
            else return i;
        }
    }
};